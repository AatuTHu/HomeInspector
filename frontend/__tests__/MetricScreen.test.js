import { render, fireEvent, waitFor } from '@testing-library/react-native'
import MetricScreen from '../components/cells/MetricsScreen'


describe('MetricsScreen', () => {
    it('should render metrics on screen', () => {
        const humidity = [{humidity: 23, id: 1, location: 'Kitchen', time: "12:00",date:"11.12"}]
        const temperature = [{temperature: 25, id: 2, location: 'Living Room', time: "13:00",date:"09.12"}]
        const screen = render(<MetricScreen humidity={humidity} temperature={temperature}/>);
        expect(screen).toMatchSnapshot();
    })

    it('should render with empty arrays', () => {
        const { getByText } = render(<MetricScreen humidity={""} temperature={""}/>)

        expect(getByText('No metric data found')).toBeTruthy();
    })

    it('should render pin and delete buttons', () => {
        const humidity = [{humidity: 23, id: 1, location: 'Kitchen', time: "12:00",date:"11.12"}]
        const temperature = [{temperature: 25, id: 2, location: 'Living Room', time: "13:00",date:"09.12"}]
        const { getAllByText } = render(<MetricScreen humidity={humidity} temperature={temperature}/>)

        expect(getAllByText("Pin")).toBeTruthy()
        expect(getAllByText("Delete")).toBeTruthy()
    })

    it('should Pin a metric if pin button is pressed', async() => {
        global.fetch = jest.fn(() =>
        Promise.resolve({
        status: 200,
        json: () => Promise.resolve({}),
        }))

        const humidity = [{humidity: 23, id: 1, location: 'Kitchen', time: "12:00",date:"11.12"}]
        const mockSetHumidity = jest.fn()
        const { getByText } = render(<MetricScreen humidity={humidity} temperature={""} setHumidity={mockSetHumidity} setTemperature={jest.fn()}/>)

        expect(getByText("Pin")).toBeTruthy()
        expect(getByText("Delete")).toBeTruthy()

        const pinButton = getByText("Pin")

        fireEvent.press(pinButton)

        await waitFor(() => {
            expect(mockSetHumidity).toHaveBeenCalled()
        })
    })

    it('should delete a metric if delete button is pressed', async() => {
        global.fetch = jest.fn(() =>
        Promise.resolve({
        status: 200,
        json: () => Promise.resolve({}),
        }))

        const humidity = [{humidity: 23, id: 1, location: 'Kitchen', time: "12:00",date:"11.12"}]
        const mockSetHumidity = jest.fn()
        const { getByText } = render(<MetricScreen humidity={humidity} temperature={""} setHumidity={mockSetHumidity} setTemperature={jest.fn()}/>)

        expect(getByText("Pin")).toBeTruthy()
        expect(getByText("Delete")).toBeTruthy()

        const deleteButton = getByText("Delete")

        fireEvent.press(deleteButton)

        await waitFor(() => {
            expect(mockSetHumidity).toHaveBeenCalled()
        })
    })
})