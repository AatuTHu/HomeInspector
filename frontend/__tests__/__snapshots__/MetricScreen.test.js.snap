// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetricsScreen should render metrics on screen 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        {
          "backgroundColor": "#FFFF",
          "flex": 1,
          "flexDirection": "column",
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "borderRadius": 22,
            "height": 50,
            "justifyContent": "center",
            "marginTop": 20,
            "width": "95%",
          }
        }
      >
        <Text
          style={
            [
              {
                "fontSize": 22,
                "textAlign": "center",
              },
            ]
          }
        >
          Temperatures
        </Text>
      </View>
      <View>
        <View
          style={
            {
              "borderStyle": "dashed",
              "borderTopWidth": 1,
              "display": "flex",
              "flexDirection": "row",
              "height": 230,
              "justifyContent": "space-evenly",
              "marginBottom": 15,
              "marginTop": 15,
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#E8F1F2",
                "borderRadius": 10,
                "borderWidth": 1,
                "display": "flex",
                "justifyContent": "space-evenly",
                "marginTop": 20,
                "width": "45%",
              }
            }
          >
            <Text
              style={
                {
                  "fontSize": 19,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              temperature
            </Text>
            <Text
              style={
                {
                  "fontSize": 15,
                  "textAlign": "center",
                }
              }
            >
              09.12
               - 
              13:00
            </Text>
            <Text
              style={
                {
                  "fontSize": 15,
                  "textAlign": "center",
                }
              }
            >
              25
               
              Â°C
            </Text>
            <Text
              style={
                {
                  "fontSize": 15,
                  "textAlign": "center",
                }
              }
            >
              Living Room
            </Text>
          </View>
          <View
            style={
              {
                "display": "flex",
                "justifyContent": "space-evenly",
                "width": "30%",
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#E3B23C",
                  "borderRadius": 20,
                  "borderWidth": 1,
                  "marginTop": 20,
                  "opacity": 1,
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#001021",
                    "fontSize": 22,
                    "textAlign": "center",
                  }
                }
              >
                Pin
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#FF6542",
                  "borderRadius": 20,
                  "borderWidth": 1,
                  "marginTop": 20,
                  "opacity": 1,
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#001021",
                    "fontSize": 22,
                    "textAlign": "center",
                  }
                }
              >
                Delete
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "borderRadius": 22,
            "height": 50,
            "justifyContent": "center",
            "marginTop": 20,
            "width": "95%",
          }
        }
      >
        <Text
          style={
            [
              {
                "fontSize": 22,
                "textAlign": "center",
              },
            ]
          }
        >
          Humidity
        </Text>
      </View>
      <View>
        <View
          style={
            {
              "borderStyle": "dashed",
              "borderTopWidth": 1,
              "display": "flex",
              "flexDirection": "row",
              "height": 230,
              "justifyContent": "space-evenly",
              "marginBottom": 15,
              "marginTop": 15,
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#E8F1F2",
                "borderRadius": 10,
                "borderWidth": 1,
                "display": "flex",
                "justifyContent": "space-evenly",
                "marginTop": 20,
                "width": "45%",
              }
            }
          >
            <Text
              style={
                {
                  "fontSize": 19,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              humidity
            </Text>
            <Text
              style={
                {
                  "fontSize": 15,
                  "textAlign": "center",
                }
              }
            >
              11.12
               - 
              12:00
            </Text>
            <Text
              style={
                {
                  "fontSize": 15,
                  "textAlign": "center",
                }
              }
            >
              23
               
              %
            </Text>
            <Text
              style={
                {
                  "fontSize": 15,
                  "textAlign": "center",
                }
              }
            >
              Kitchen
            </Text>
          </View>
          <View
            style={
              {
                "display": "flex",
                "justifyContent": "space-evenly",
                "width": "30%",
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#E3B23C",
                  "borderRadius": 20,
                  "borderWidth": 1,
                  "marginTop": 20,
                  "opacity": 1,
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#001021",
                    "fontSize": 22,
                    "textAlign": "center",
                  }
                }
              >
                Pin
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#FF6542",
                  "borderRadius": 20,
                  "borderWidth": 1,
                  "marginTop": 20,
                  "opacity": 1,
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#001021",
                    "fontSize": 22,
                    "textAlign": "center",
                  }
                }
              >
                Delete
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
